Machine.Specifications.ReSharper.Provider.10/MspecServiceProvider.cs

```diff
@@ -59,5 +64,14 @@ public UnitTestElementId CreateId(IProject project, TargetFrameworkId targetFram
         {
             return _elementIdFactory.Create(_provider, project, targetFrameworkId, id);
         }
+
+        private void AddElementHandler(Lifetime lifetime)
+        {
+            ElementManager.UnitTestElementsRemoved.Advise(lifetime, set =>
+            {
+                foreach (var element in set)
+                    ElementManager.RemoveElements(element.Children.ToSet());
+            });
```

Comment:
Had to add this handler to remove any child elements of containers that get removed. For some reason, behavior specs lingered even though the parent context or behavior were removed from the tree.